llvm/include/llvm/CodeGen/MachineInstr.h

```diff
@@ -114,6 +114,7 @@ class MachineInstr
                              // this instruction.
     Unpredictable = 1 << 16, // Instruction with unpredictable condition.
     NoConvergent = 1 << 17,  // Call does not require convergence guarantees.
+    Fusible = 1 << 18,       // Instruction is the second of a fusible pair.
```

Comment:
Is the point to add a flag to simplify the code which detects fusible patterns? I realise that some processors may have a lot of fusions, but is adding a new flag really the right solution?

Reply:
> Is the point to add a flag to simplify the code which detects fusible patterns? 

Yes!

> I realise that some processors may have a lot of fusions, but is adding a new flag really the right solution?

I think this can be feasible, there are already some values in MIFlag that are not so common, like `NoConvergent` (SIMT only I think), `Unpredictable` (X86 only), etc.

Reply:
What if the second fusion instruction is an instruction with 2 register sources instead of 1 register and an immediate. We'll need to know which operand to constrain.

Reply:
> What if the second fusion instruction is an instruction with 2 register sources

Currently, I think this is a uncommon case and I don't know if there are such fusions. Do you have any example? Maybe integer multiply-add fusion?
Even if so, I think we can know the constrainted operands via its opcode and constrain them in the `getRegAllocationHints` implementation. We don't need to record this info in target-independent part.

Reply:
From the RISC-V section of this page https://en.wikichip.org/wiki/macro-operation_fusion which are also listed here https://xiangshan-doc.readthedocs.io/zh-cn/latest/frontend/decode/

```
slli rd, rs1, {1,2,3}
add rd, rd, rs2
```

There are a few others on https://xiangshan-doc.readthedocs.io/zh-cn/latest/frontend/decode/

Reply:
Thanks, I didn't take a deep look into XiangShan's fusions (facepalming).